packages/meditrak-server/src/dhis/DhisChangeSideEffectHandler.js

```diff
@@ -21,9 +21,12 @@ export class DhisChangeSideEffectHandler extends ChangeSideEffectHandler {
           SELECT DISTINCT survey_response.*
           FROM survey_response
           JOIN survey ON survey_response.survey_id = survey.id
+          JOIN data_source ds ON survey.data_source_id = ds.id
           JOIN entity ON survey_response.entity_id = entity.id
-          WHERE survey_response.id IN (${batchOfSurveyResponseIds.map(() => '?').join(',')})
-          AND ${this.models.surveyResponse.getOnlyEventsQueryClause()};
+          WHERE
+            survey_response.id IN (${batchOfSurveyResponseIds.map(() => '?').join(',')})
+            AND ${this.models.surveyResponse.getOnlyEventsQueryClause()}
+            AND ds.service_type = 'dhis';
```

Comment:
@edmofro  We have missed this in the past - we testing this locally, I was wondering why changed responses were always processed twice (they were marked as changed by this function)

Reply:
Right! Good catch, thanks for tidying up 🙏